src/gpu/sycl/ref_layer_normalizations.cpp

```diff
@@ -0,0 +1,222 @@
+                DNNL_ARG_ATTR_SCALES | DNNL_ARG_DST);
+
+        auto var = CTX_IN_SYCL_KERNEL_MEMORY(DNNL_ARG_VARIANCE);
+        auto var_out = CTX_OUT_SYCL_KERNEL_MEMORY(DNNL_ARG_VARIANCE);
+        auto dst = CTX_OUT_SYCL_KERNEL_MEMORY(DNNL_ARG_DST);
+        auto nelems_A = memory_desc_wrapper(pd()->src_md(0)).nelems();
+
+        layer_normalization_fwd_kernel_vec_t layer_normalization_fwd_kernel(
+                pd()->conf_, data, scale, shift, mean, var, dst, mean_out,
+                var_out, src_scales, dst_scales);
+
+        const int block_size = pd()->conf_.block_size;
+        const int wg_size = pd()->conf_.wg_size;
+
+        int work_per_wg = wg_size * block_size;
+        int n_wgs = (pd()->across_axis() + work_per_wg - 1) / work_per_wg;
+        int n_thr = n_wgs * wg_size;
+
+        cgh.parallel_for(::sycl::nd_range<1>(n_thr, wg_size),
+                layer_normalization_fwd_kernel);
+    });
+
+    return status::success;
+}
+
+status_t ref_layer_normalization_bwd_t::pd_t::init_conf() {
+    conf_ = sycl_layer_normalization_conf_t();
+
+    conf_.var_md = sycl_md_t(src_md(2));
+    conf_.ndims = ndims();
+    conf_.flags = desc()->flags;
+    conf_.block_size = (16);
+    conf_.wg_size = (32);
+    conf_.prop_kind = desc_.prop_kind;
+    conf_.use_scale = use_scale();
+    conf_.use_shift = use_shift();
+    conf_.data_md = sycl_md_t(src_md(0));
+    conf_.diff_data_md = sycl_md_t(diff_src_md(0));
+    conf_.data_scaleshift_md = sycl_md_t(weights_md(0));
+    conf_.diff_data_scaleshift_md = sycl_md_t(diff_weights_md(0));
+    conf_.stat_md = sycl_md_t(src_md(1));
+    conf_.dst_md = sycl_md_t(dst_md(0));
+    conf_.diff_dst_md = sycl_md_t(diff_dst_md(0));
+    conf_.stat_d = sycl_md_t(stat_md());
+    conf_.zero_dims = has_zero_dim_memory();
+    auto nelems_A = memory_desc_wrapper(src_md(0)).nelems();
+    conf_.diff_shift_off = conf_.use_ss && !conf_.zero_dims
+            ? conf_.diff_data_scaleshift_md.off(1, 0)
+            : 0;
+    int work_per_wg = conf_.wg_size * conf_.block_size;
+    int n_wgs = (nelems_A + work_per_wg - 1) / work_per_wg;
+    int n_wgs2 = (across_axis() + work_per_wg - 1) / work_per_wg;
+    conf_.n_thr = n_wgs * conf_.wg_size;
+    conf_.zero_dims = has_zero_dim_memory();
+    conf_.N = across_axis();
+    conf_.C = norm_axis();
+    conf_.layer_norm_epsilon = desc()->layer_norm_epsilon;
+    conf_.save_stats = is_training();
+    conf_.calculate_diff_stats = !use_global_stats();
+    ;
+
+    return status::success;
+}
+
+status_t ref_layer_normalization_bwd_t::init(engine_t *engine) {
+    if (pd()->use_scale() || pd()->use_shift()) {
+        const auto kid
+                = ::sycl::get_kernel_id<layer_normalization_bwd_kernel_vec_t>();
+        CHECK(create_kernel(engine, kid, &kernel_));
+    }
+    const auto kid
+            = ::sycl::get_kernel_id<layer_normalization_bwd_kernel_vec_t2>();
+    CHECK(create_kernel(engine, kid, &kernel_));
+    return status::success;
```

Comment:
Can you please explain logic behind this code? There is only 1 kernel named `kernel_` and in the case when we need a scale or shift we initialize `kernel_` with `layer_normalization_bwd_kernel_vec_t` but then we overwrite it by initializing it once again with `layer_normalization_bwd_kernel_vec_t2`.

Reply:
Any update on this? I see that the code hasn’t changed.

Reply:
Fixed kernel overwriting